### PR TITLE
Fix #368

### LINKED ISSUE #368: Application crash when closing Floating Editor and Docked Editor in Demo
Operating System: Kubuntu 21.04
KDE Plasma Version: 5.21.4
KDE Frameworks Version: 5.80.0
Qt Version: 5.15.2
Kernel Version: 5.11.0-38-generic
OS Type: 64-bit
Graphics Platform: X11

Steps:

1. Run AdvancedDockingSystemDemo
2. Create Docked Editor
3. Click to close created Docked Editor
4. Application crash after calling the QApplication::focusWindowChanged signal handler  CDockFocusController::onFocusWindowChanged. The dynamic property "FocusedDockWidget" refers to an already deleted object.

### PR DESCRIPTION
**Pull request by gorec323**
Application crash when closing Floating Editor and Docked Editor in Demo #368

### CODE REVIEW

**Comment by githubuser0xFFFF** [comment]
First I would like to thank you for your pull request and for fixing an issue.

At the moment I cannot accept the pull request as it is, because you mix several changes and commit that are not related to the problem that you try to fix in one single pull request. This makes it harder for me to review and test the code and this also increases the probability of unwanted side effects.

Please create a clean pull request, that fixes one single issue. Move other changes (cmake.yml and README.md) into separate pull reuests.

Thank you.

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Advanced Docking System for Qt
 
-[![Build status](https://github.com/githubuser0xFFFF/Qt-Advanced-Docking-System/workflows/linux-builds/badge.svg)](https://github.com/githubuser0xFFFF/Qt-Advanced-Docking-System/actions?query=workflow%3Alinux-builds)
+[![linux-builds](https://github.com/gorec323/Qt-Advanced-Docking-System/actions/workflows/linux-builds.yml/badge.svg)](https://github.com/gorec323/Qt-Advanced-Docking-System/actions/workflows/linux-builds.yml)
```
[the remark this answers is not in the document]
Comment (githubuser0xFFFF) [reply README.md:3]: If I merge this, then I will link the batches in README.md to your workflow in your GitHub account

**Review by githubuser0xFFFF — COMMENTED** [review]
First I would like to thank you for your pull request and for fixing an issue.

At the moment I cannot accept the pull request as it is, because you mix several changes and commits that are not related to the problem that you try to fix in one single pull request. This makes it harder for me to review and test the code and this also increases the probability of unwanted side effects.

Please create a clean pull request, that fixes one single issue. Move other changes (cmake.yml and README.md) into separate pull reuests.

Thank you.

**Comment by gorec323** [comment]
I have created a new pull request that contains only those commits that fix this bug.

**Pull request closed without merging**